ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -452,14 +457,18 @@ decodeShelleyResult query = case query of
     GetEpochNo                                 -> decode
     GetNonMyopicMemberRewards {}               -> decode
     GetCurrentPParams                          -> fromCBOR
-    GetProposedPParamsUpdates                  -> fromCBOR
+    GetProposedPParamsUpdates                  -> unsafeForHashingPPDeltaDecoder
     GetStakeDistribution                       -> fromCBOR
     GetFilteredUTxO {}                         -> fromCBOR
     GetUTxO                                    -> fromCBOR
-    DebugEpochState                            -> fromCBOR
+    DebugEpochState                            -> forgetAnnotator fromCBOR
     GetCBOR {}                                 -> decode
     GetFilteredDelegationsAndRewardAccounts {} -> fromCBOR
     GetGenesisConfig                           -> fromCBOR
-    DebugNewEpochState                         -> fromCBOR
+    DebugNewEpochState                         -> forgetAnnotator fromCBOR
     DebugChainDepState                         -> fromCBOR
     GetRewardProvenance                        -> fromCBOR
+    where
+       forgetAnnotator ::
+         forall a r. (forall t. Decoder t (Annotator a)) -> Decoder r a
+       forgetAnnotator d = (\x -> runAnnotator x (Full mempty)) <$> d
```

Comment:
This only works here because the `FromCBOR` instance for `Annotator NewEpochState` doesn't actually use the annotations. This is apparently recursively true for its fields too.

/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs#L675-L684

Here is an example of a `FromCBOR` that does use the annotation:

https://github.com/input-output-hk/cardano-ledger-specs/blob/5b6ab459841b59432400b5d0cb796c2a12344bf9/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Tx.hs#L408

Reply:
Ah. Shelley-based `NewEpochState` contains `EpochState` which contains `PParams` which, even for `Alonzo`, does not contain any annotations. That's why we can use our trick here. (Well, because _none_ of the fields in `NewEpochState` carry annotations. EG There are no `MemoBytes` anywhere in there.)

The annotations are only present in `LangDepView`, which is created by calling `getLanguageView` which ultimately (via `To`) depends on a `ToCBOR` instance.

Reply:
Which is a long-winded way of emphasizing that `forgetAnnotator` relies on special assumptions: that the `FromCBOR (Annotator NewEpochState)` instance is rather degenerate: the `Annotator` layer is (at least currently) completely unnecessary.

Reply:
gotcha! I hope to add back the relevant `FromCBOR` instances soon and then come clean this back up. Assuming we all agree that it is okay to have the node provide the serialization of the protocol parameters. Otherwise we will have to handle the annotation correctly. Thanks again for the help!